File: python/mycompany.py
class MyCompany(object):
    '''
    Let access different entities of mycompany.com
    '''

    url = 'mycompany.com'

    def projects(self):
        return self._get('/projects')

    def employees(self):
        return self._get('/employees')

    def customers(self):
        return self._get('/customers')

    def products(self):
        return self._get('/products')

    def _get(self, path):
        return 'GET ' + self.url + path

File: javascript/mycompany.js

function MyCompany() {
    this.url = 'mycompany.com';
};

MyCompany.prototype.projects = function() {
    return this._get('/projects');
}

MyCompany.prototype.employees = function() {
    return this._get('/employees');
}

MyCompany.prototype.customers = function() {
    return this._get('/customers');
}

MyCompany.prototype._get = function(path) {
    return 'GET ' + this.url + path;
}
